js/main.js

```diff
@@ -135,5 +134,45 @@ ul.addEventListener('click', function (event) {
       previewPhoto.setAttribute('src', prePopulate.$url);
     }
   }
+  deleteButton();
   document.querySelector('#formTitle').innerHTML = 'Edit Entry';
 });
+
+const del = document.querySelector('.delete');
+function deleteButton() {
+  del.classList.remove('hidden');
+}
+
+const mod = document.querySelector('.modal');
+del.addEventListener('click', function () {
+  mod.classList.remove('hidden');
+});
+
+const cancel = document.querySelector('.cancel');
+cancel.addEventListener('click', function () {
+  mod.classList.add('hidden');
+});
+
+const confirm = document.querySelector('.confirm');
+confirm.addEventListener('click', function () {
+  function confirmBtn(array) {
+    for (let i = 0; i < data.entries.length; i++) {
+      if (data.editing.entryId !== data.entries[i].entryId) {
+        data.entries = data.entries[i];
+      }
+    }
+  }
+  const filtered = data.entries.filter(confirmBtn);
+  data.entries = filtered;
+  const liElements = document.querySelectorAll('li');
+  for (let j = 0; j < liElements.length; j++) {
+    if (data.entries.entryId === liElements[j].getAttribute('data-entry-id')) {
+      liElements[j].remove();
+    }
+  }
+  if (data.entries.length === 0) {
+    toggleNoEntries();
+  }
+  mod.classList.add('hidden');
+  viewSwap('entries');
```

Comment:
after line 177, add a call to the `resetForm` and also make sure to set `data.editing` to `null`. Otherwise you'll run into a bug where attempting to add/delete subsequent entries doesn't work.